### ISSUE TITLE
Can't load MathJax 3 extensions

### ISSUE THREAD

**Issue by TheoFletcher** [doxygen/doxygen#9217]
I'm unable to properly load various MathJax extensions with version 1.9.3 and the configuration:
```
MATHJAX_VERSION        = MathJax_3
MATHJAX_RELPATH        = https://cdn.jsdelivr.net/npm/mathjax@3
MATHJAX_EXTENSIONS     = extension1 extension2 ...
```
where `extension?` is an extension as listed in the MathJax documentation, http://docs.mathjax.org/en/latest/input/tex/extensions/index.html .

For example, with `MATHJAX_EXTENSIONS = boldsymbol` the input `/** @f$ \boldsymbol{A} @f$ */` gives the error "Undefined control sequence \boldsymbol", and similar for other extensions (mhchem, physics, ...).

I found that using `MATHJAX_EXTENSIONS = autoload` will fix this in most cases, but won't work with "ams" or "physics".

Another work-around is to use the require extension:
`MATHJAX_EXTENSIONS = require`
 `/** @f$ \require{boldsymbol} \boldsymbol{A} @f$ */`

From looking at https://github.com/doxygen/doxygen/blob/8c0906e8205764b2736bcc849b5b5ea707a34d8f/templates/html/htmlbase.tpl , I think the issue is that user defined extensions aren't actually being loaded, and so only the ones that are automatically loaded (including autoload and require) are working.

**Comment by albert-github** [comment]
The fact that the plain `boldsymbol` doesn't work I can reproduce and this will go away by:
- using `autoload` as indicated
- adding `loader: {load: ['[tex]/boldsymbol']},`

I think doxygen should add the `loader: {load: ['[tex]/boldsymbol']},` version.

The fact that `autoload` doesn't work for `ams` and `physics` makes the automatic use of `autoload` not desirable.
The reason that `autoload` doesn't work is, as @TheoFletcher hinted, at  in the documentation:
> The autoload extension predefines all the macros from the extensions that haven’t been loaded already so that they automatically load the needed extension when they are first used, with the exception of the [physics](http://docs.mathjax.org/en/latest/input/tex/extensions/physics.html#tex-physics) package, since it redefines standard macros, and the [ams](http://docs.mathjax.org/en/latest/input/tex/extensions/ams.html#tex-ams) package, due to the large number of macros it contains.


It is a bit strange that that `ams` doesn't work as this (I) was / should have been tested during development.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9218

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
